packages/core/core/src/parcel-v3/FileSystem.js

```diff
@@ -0,0 +1,27 @@
+// @flow
+
+import type {FileSystem} from '@parcel/types';
+import type {HandlerFunc, RpcEventRouter} from './RpcEventRouter';
+
+//
+// Events Defined in ~/crates/parcel_plugin_rpc/src/nodejs/file_system/file_system.rs
+//
+type ReadToStringHandler = HandlerFunc<'fs/read_to_string', string, string>;
+type IsFileHandler = HandlerFunc<'fs/is_file', string, boolean>;
+type IsDirHandler = HandlerFunc<'fs/is_dir', string, boolean>;
+
+export class FileSystemRpc {
```

Comment:
Feel like a function would be a cleaner choice here.

Reply:
I think so too, though I am wondering if we need to have methods on it - perhaps a `.destroy()` to detach listeners